frontend/web/components/SegmentOverrides.js

```diff
@@ -175,48 +175,46 @@ const SegmentOverrideInner = class Override extends React.Component {
                 />
               )}
               <Row className='gap-2'>
-                {!!v.id && (
-                  <Permission
-                    id={projectId}
-                    permission={'MANAGE_SEGMENTS'}
-                    level={'project'}
-                  >
-                    {({ permission }) =>
-                      Utils.renderWithPermission(
-                        permission,
-                        Constants.projectPermissions('Manage Segments'),
-                        <>
-                          {v.is_feature_specific ? (
-                            <Button
-                              disabled={!permission}
-                              onClick={() => {
-                                setShowCreateSegment(true)
-                                setSegmentEditId(v.segment)
-                              }}
-                              className='btn btn-with-icon'
-                            >
-                              <span className='no-pointer'>
-                                <Icon name='edit' fill={'#656D7B'} width={20} />
-                              </span>
-                            </Button>
-                          ) : (
-                            <Button
-                              theme='text'
-                              disabled={!permission}
-                              target='_blank'
-                              href={`${document.location.origin}/project/${this.props.projectId}/environment/${this.props.environmentId}/segments?id=${v.segment}`}
-                              className='btn btn-with-icon'
-                            >
-                              <span className='no-pointer'>
-                                <Icon name='edit' fill={'#656D7B'} width={20} />
-                              </span>
-                            </Button>
-                          )}
-                        </>,
-                      )
-                    }
-                  </Permission>
-                )}
+                <Permission
```

Comment:
Prior to this you couldn't go back and edit a newly created feature-based segment

<img width="663" alt="image" src="https://github.com/Flagsmith/flagsmith/assets/8608314/7943af27-4167-4659-866d-91f56a07ba8e">


<img width="645" alt="image" src="https://github.com/Flagsmith/flagsmith/assets/8608314/3a448f79-d7d4-496c-ab98-7738db1da4e9">